### ISSUE TITLE
Optimal the HTML format

### ISSUE THREAD

**Issue by tmdfree** [michaelfm1211/mdview#2]
The readability of output HTML is not good. If add '\n' after every begin and end label, it will be fine. 

markdown file
```
# Head Level 1

        Hello Markdown.

## Head Level 1
### Head Level 1
## Head Level 1
```
orign output HTML
```
<h1 id="1"> Head Level 1</h1><p>        Hello Markdown. </p><h2 id="2"> Head Level 1</h2><h3 id="3"> Head Level 1</h3><h2 id="4"> Head Level 1</h2>
```

new ouotput HTML

```
<h1 id="1">
        Head Level 1
</h1>
<p>
        Hello Markdown.
</p>
<h2 id="2">
Head Level 1
</h2>
<h3 id="3">
Head Level 1
</h3>
<h2 id="4">
Head Level 1
</h2>

```

Of course, as following is best. But it seems more difficult to code.

```
<h1 id="1">
        Head Level 1
</h1>
<p>
        Hello Markdown.
</p>
        <h2 id="2">
        Head Level 1
        </h2>
                <h3 id="3">
                Head Level 1
                </h3>
        <h2 id="4">
        Head Level 1
        </h2>

```

**Comment by michaelfm1211** [comment]
> The readability of output HTML is not good

I agree. Making it prettier has been on my mind for a while, but I never considered it a priority because compiler output isn't really meant to be hand-edited. Still making it prettier is just a nice touch in general and makes debugging easier.

Perfectly prettifying auto-generated stuff is a bit hard, so I don't think too much attention should be put here. If prettiness is really important for you, then I suggest just running the output through [`tidy`](https://github.com/htacg/tidy-html5) (though it might slow down your build). There is a libtidy too which could be integrated into the `mdv` frontend which might be useful (AFAIK it can't be used on the backend because the backend doesn't know the full Markdown at any given time).

> If add '\n' after every begin and end label, it will be fine.

I've implemented a version of this in 0c780e6. Starting tags for block elements are followed by new lines, and closing tags for block elements are preceded and followed by new lines. The exception is header tags and list items (`<li>`): their starting tags are the same but their closing tags are only followed by new lines. The result is this:
```
# Head Level 1

        Hello Markdown.

## Head Level 1
### Head Level 1
## Head Level 1
```
compiles to
```
<h1 id="1"> Head Level 1</h1>
<p>
        Hello Markdown. 
</p>
<h2 id="2"> Head Level 1</h2>
<h3 id="3"> Head Level 1</h3>
<h2 id="4"> Head Level 1</h2>
```

**Comment by tmdfree** [comment]
Both [tidy](https://github.com/htacg/tidy-html5) and https://github.com/michaelfm1211/mdview/commit/0c780e6a6c84f45096c2fd7f5c1f0878c131bb25 work. 

Thanks for your code and advies. It is very helpful to me.
